Core/src/org/sleuthkit/autopsy/modules/vmextractor/VMExtractorIngestModule.java

```diff
@@ -237,6 +239,46 @@ public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress
         }
         return vmFiles;
     }
+    
+    /**
+     * Check all the files and if a file is a vhd then check to make sure it is a valid vhd using the mimetype. We are not
+     * checking the mimetype for VMDK's at this point in time.
+     * 
+     * @param vmFiles List of virtual machine abstract files to look at
+     * 
+     * @return List of abstract files of virtual machine files.
+     */
+    private static List<AbstractFile> removeNonVMFiles(List<AbstractFile> vmFiles) {
+        List<AbstractFile> vFile = new ArrayList<>();
+
+        for (AbstractFile vmFile : vmFiles) {
+            if (vmFile.getNameExtension().equalsIgnoreCase("vhd")) {
+                String fileMimeType = vmFile.getMIMEType();
+                if (fileMimeType == null) {
+                    FileTypeDetector fileTypeDetector = null;
+                    try {
+                        fileTypeDetector = new FileTypeDetector();
```

Comment:
Can we make one FileTypeDetector at the beginning and reuse it?

Reply:
Yes, done.